src/v/model/fundamental.h

```diff
@@ -200,6 +200,10 @@ inline constexpr model::offset prev_offset(model::offset o) {
     return o - model::offset{1};
 }
 
+// An invalid offset indicating that actual LSO is not yet ready to be returned.
+// Follows the policy that LSO is the next offset of the decided offset.
+static constexpr model::offset invalid_lso{next_offset(model::offset::min())};
```

Comment:
It will be 0.

```
        auto maybe_lso = _partition->last_stable_offset();
        if (maybe_lso == model::invalid_lso) {
            return error_code::offset_not_available;
        }
```
Can `last_stable_offset` return 0 in case when it is real last stable offset, not indicator of error?